rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py

```diff
@@ -24,6 +25,20 @@
 from rosidl_parser import validate_field_types
 
 
+def check_idlpp_supports_include_namespaces(idl_pp):
+    r = subprocess.getstatusoutput(idl_pp + ' -v')
+    ospl_version = r[1].split(':')[1].strip()
+    m = re.search('([1-9][0-9]*\.[0-9]+)\.([0-9]*)', ospl_version)
+    if m and m.lastindex == 2:
+        major = m.group(1)
+        minor = m.group(2)
+
+    if ospl_version.endswith('OSS'):
+        return major > '6.9' or (major == '6.9' and minor > '181127')
```

Comment:
This comparisons of the `major` variable are incorrect: e.g. `major > '6.9'`

For `major` being `6.10` the expression evaluates to `False` where it should be `True.

Reply:
As an alternative we could introduce a version dependency to require the latest release when backporting this patch to Crystal? That would avoid the need for the version check and always use `-o maintain-include-namespace`.

Reply:
Please see #21 for the described alternative. What do you think?